File: test-data/typescript/features/decorators.ts
function log(target: any, propertyKey: string, descriptor: PropertyDescriptor) {
    const originalMethod = descriptor.value;
    descriptor.value = function (...args: any[]) {
        console.log(`Calling ${propertyKey} with`, args);
        const result = originalMethod.apply(this, args);
        console.log(`Result: ${result}`);
        return result;
    };
    return descriptor;
}

export class Service {
    @log
    process(data: string): string {
        return data.toUpperCase();
    }
}

File: test-data/python/features/decorators.py
import functools

def log_execution(func):
    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        print(f"Executing {func.__name__}")
        result = func(*args, **kwargs)
        print(f"Finished {func.__name__}")
        return result
    return wrapper

class Service:
    @log_execution
    def process(self, data: str) -> str:
        return data.upper()
